### DISCUSSION TITLE
Error loading saved models in Google Colab Pro

### DISCUSSION THREAD

**Discussion by rcmoraleshernandez** [MaartenGr/BERTopic#1125] [Q&A]
Hi Maarten,

I'm using Google Colab pro, I made and save BERTopic models since last year, but now I can **not** read any saved model. The errors looks like is in PICKLE... see the images below of 2 attempts to load different models.
I even install BERTopic ver 12.0 to try to read 'old' models saved with this version, but I can't read the model either.

Is there anything I can do?
I really appreciate your time and support.

#####CODE

!pip install bertopic
>>>bertopic in /usr/local/lib/python3.9/dist-packages (0.14.1)

from enum import auto
from bertopic import BERTopic

topic_model = BERTopic.load("/content/drive/MyDrive/Colab Notebooks/....../model_03mar23")

![error loading bert model](https://user-images.githubusercontent.com/73208310/227648841-2534e726-c445-4d6d-8b81-bfdd32e859b8.jpg)

![2error loading bert model](https://user-images.githubusercontent.com/73208310/227647647-64fcea30-d0ce-481d-854d-28d7554b8b17.jpg)

**Comment by MaartenGr** [comment]
When you save a model in BERTopic, and the same applies to [scikit-learn](https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations), it is important to keep track of your dependencies as well as the python version that you are using. For example, even if you were to set BERTopic to 0.12, that does not guarantee that your environment is exactly the same as it was before. Dependencies might have updated in between. Moreover, Google Colab updates its python version once in a while which also changes the environment in which you work. If you performed version control by tracking versions of dependencies and python, then you revert the environment to what you tracked, and loading the model should work.
